packages/flutter_tools/test/general.shard/version_test.dart

```diff
@@ -144,296 +141,7 @@ void main() {
       }, overrides: <Type, Generator>{
         FlutterVersion: () => FlutterVersion(clock: _testClock),
         ProcessManager: () => processManager,
-        Cache: () => mockCache,
-      });
-
-      testUsingContext('prints nothing when Flutter installation looks out-of-date but is actually up-to-date', () async {
```

Comment:
Should these be replaced?

Reply:
Normally we do have unit tests which provide fake commands and make assertions based on that output. What I see here is 99% mocks in order to test functionality - correctly delaying version check updates - which is not that important. 

The way I think we could test it is by taking https://github.com/flutter/flutter/blob/e7b7ebc066c1b2a5aa5c19f8961307427e0142a6/packages/flutter_tools/lib/src/version.dart#L359 and breaking it out of the version class so we can make assertions based on whether the local/remote is up to date without specifying the exact git commands.

Reply:
I think I can do that here